onchain/src/tests/art_peace.cairo

```diff
@@ -425,3 +425,49 @@ fn deposit_reward_test() {
         art_peace_token_balance == reward_amount, "reward wrongly distributed when adding template"
     );
 }
+
+#[test]
+fn distribute_rewards_test() {
+    let art_peace_address = deploy_contract();
+    let art_peace = IArtPeaceDispatcher { contract_address: art_peace_address };
+    let template_verifier = ITemplateVerifierDispatcher { contract_address: art_peace.contract_address };
+    let template_store = ITemplateStoreDispatcher { contract_address: art_peace.contract_address };
+
+    let erc20_mock: ContractAddress = deploy_erc20_mock();
+    let reward_amount: u256 = 1 * utils::pow_256(10, 18);
+
+    let template_image = array![1, 2, 3, 4];
+    let template_hash = compute_template_hash(template_image.span());
+    let template_metadata = TemplateMetadata {
+        name: 'test',
+        hash: template_hash,
+        position: 0,
+        width: 2,
+        height: 2,
+        reward: reward_amount,
+        reward_token: erc20_mock,
+        creator: get_caller_address(),
+    };
+
+    let template_image_span = template_image.span();
+    let x = 10;
+    let y = 20;
+    let pos = x + y * WIDTH;
+    let color = 0x5;
+    let now = 10;
+    let template_id = 0;
+
+    IERC20Dispatcher { contract_address: erc20_mock }.approve(art_peace_address, reward_amount);
+    template_store.add_template(template_metadata);
+    assert!(template_store.get_templates_count() == 1, "Templates count is not 1");
+    art_peace.place_pixel(pos, color, now);
```

Comment:
You should need to place all 4 template image pixels to be able to complete the template properly